dist/backdraft.js

```diff
@@ -1476,6 +1544,70 @@ _.extend(Plugin.factory, {
       this.trigger("draw", arguments);
     },
 
+    exportData : function(sUrl) {
+      var oSettings = this.dataTable.fnSettings;
+      var aoData = this.dataTable._fnAjaxParameters( oSettings );
+      this._addServerParams( aoData );
+      this._fetchCSV( sUrl, aoData );
+    },
+
+    _fetchCSV : function (sUrl, aoData) {
+      if (this.serverSideFiltering) {
+        var filterJson = {};
+        filterJson.name = "ext_filter_json";
+        filterJson.value = this._getFilteringSettings();
+        aoData.push(filterJson);
+      }
+      $.ajax({
+        url: sUrl,
+        data : aoData,
+        dataType : "text",
+        cache : false,
+        type : this.ajaxMethod || "GET",
+        success: function(response, status, xhr) {
```

Comment:
we should discuss directly here.. we do this for the old tables.  The point is you send a URL over with all the correct sorting, filtering, etc and the server will respond with the CSV file.  There is nothing you are "getting out of the current table" other than what filters and option params are set.